javalib/src/main/scala/java/lang/System.scala

```diff
@@ -49,6 +51,26 @@ object System {
       sysProps.setProperty("user.country", userCountry)
 
     } else {
+      if (Platform.isMac) {
```

Comment:
I was not very clear on the previous comment, but I think this branch specifically should rather be in the C code. So that on Scala side we're left with something along the lines of:

```
        val name, tmpdir, version = ...
        Platform.osVersion(major, minor, patch)
        Platform.osName(name)
        Platform.osTmpDir(tmpdir)
        sysProps.setProperty("os.name", ...)
        sysProps.setProperty("os.version", ...)
        sysProps.setProperty("java.io.tmpdir", ...)
```
We should try avoiding platform-dependant branching in Scala code as much as possible (and keep those in C code instead).

Reply:
Yes, we started that platform branching early on and with only a couple of items and platforms it looked reasonable. Now, I agree that we should hide the platform details in the main code line.